ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -97,6 +95,75 @@ public void onUnityAdsFailedToLoad(String placementId) {
     }
   };
 
+  /**
+   * IUnityAdsExtendedListener instance for call backs from show method only.
+   */
+  private IUnityAdsExtendedListener mUnityShowListener = new IUnityAdsExtendedListener() {
+    @Override
+    public void onUnityAdsClick(String s) {
+      Log.v(TAG, "Unity interstitial ad for placement ID '" + mPlacementId + "' was clicked.");
+      // Unity Ads ad clicked.
+      if (mMediationInterstitialListener == null) {
+        return;
+      }
+      mMediationInterstitialListener.onAdClicked(UnityAdapter.this);
+      // Unity Ads doesn't provide a "leaving application" event, so assuming that the
+      // user is leaving the application when a click is received, forwarding an on ad
+      // left application event.
+      mMediationInterstitialListener.onAdLeftApplication(UnityAdapter.this);
```

Comment:
is there any return to app event that needs to be handled?

Reply:
no, we only need to trigger meditionListeners here.